### PR TITLE
Align on a single predicate type

### PR DESCRIPTION
**Pull request by aaronbuchwald**
## Why this should be merged

This PR aligns precompile predicates on a single type instead of separating out into with/without the ProposerContext.

## How this works

as above

## How this was tested

CI

## How is this documented

na

### CODE REVIEW

File: precompile/precompileconfig/config.go [hunk 1]
```diff
@@ -32,46 +32,26 @@ type Config interface {
...
-// a precompile predicate within a specific ProposerVM wrapper.
-type ProposerPredicateContext struct {
-	PrecompilePredicateContext
+// PredicateContext is the context passed in to the Predicater interface to verify
+// a precompile predicate within a specific ProposerVM header.
+type PredicateContext struct {
+	SnowCtx *snow.Context
 	// ProposerVMBlockCtx defines the ProposerVM context the predicate is verified within
 	ProposerVMBlockCtx *block.Context
 }
```
Comment (anusha-ctrl) [inline precompile/precompileconfig/config.go:53]: It would be helpful if you could add more comments for `PredicateContext` saying when you expect certain fields to be nil or not.

File: precompile/precompileconfig/config.go [hunk 2]
```diff
@@ -32,46 +32,26 @@ type Config interface {
...
-// ProposerPredicateContext is the context passed in to the ProposerPredicater interface to verify
-// a precompile predicate within a specific ProposerVM wrapper.
-type ProposerPredicateContext struct {
-	PrecompilePredicateContext
+// PredicateContext is the context passed in to the Predicater interface to verify
+// a precompile predicate within a specific ProposerVM header.
+type PredicateContext struct {
+	SnowCtx *snow.Context
 	// ProposerVMBlockCtx defines the ProposerVM context the predicate is verified within
 	ProposerVMBlockCtx *block.Context
```
Comment (anusha-ctrl) [inline precompile/precompileconfig/config.go:52]: It would be helpful if you could add more comments for PredicateContext saying when you expect certain fields to be nil or not.

File: precompile/contract/interfaces.go [hunk 3]
```diff
@@ -51,19 +51,24 @@ type AccessibleState interface {
...
 	Number() *big.Int
 	Timestamp() uint64
 }
 
+// BlockContext defines the interface to provide information about the block to a precompile
+// during the execution of the EVM.
+type BlockContext interface {
+	ConfigurationBlockContext
+}
+
```
Comment (anusha-ctrl) [inline precompile/contract/interfaces.go:65]: As @darioush said should we separate these changes for another pr and let this one focus on the predicates

File: core/predicate_check.go [hunk 4]
```diff
@@ -24,46 +24,15 @@ func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.Prop
...
-		}
-		precompileAddressChecks[address] = struct{}{}
-		predicateBytes := predicateutils.HashSliceToBytes(accessTuple.StorageKeys)
-		if err := predicater.VerifyPredicate(predicateContext, predicateBytes); err != nil {
-			return fmt.Errorf("predicate %s failed verification for tx %s: %w", address, tx.Hash(), err)
-		}
-	}
-
-	return nil
+	return checkPrecompilePredicates(rules, predicateContext, tx)
```
Comment (darioush) [inline core/predicate_check.go:27]: nit: should we inline this?

File: core/predicate_check.go [hunk 5]
```diff
@@ -24,46 +24,15 @@ func CheckPredicates(rules params.Rules, predicateContext *precompileconfig.Prop
...
 
-func checkProposerPrecompilePredicates(rules params.Rules, predicateContext *precompileconfig.ProposerPredicateContext, tx *types.Transaction) error {
+func checkPrecompilePredicates(rules params.Rules, predicateContext *precompileconfig.PredicateContext, tx *types.Transaction) error {
 	// Short circuit early if there are no precompile predicates to verify
-	if len(rules.ProposerPredicates) == 0 {
+	if len(rules.Predicates) == 0 {
 		return nil
 	}
-	precompilePredicates := rules.ProposerPredicates
+	precompilePredicates := rules.Predicates
```
Comment (darioush) [inline core/predicate_check.go:35]: nit: could we get rid of this variable (or rename to predicates, but doesn't seem very helpful)?

**Review by darioush — APPROVED** [review]

**Review by anusha-ctrl — APPROVED** [review]
